src/lib/fstab/tsort.rb

```diff
@@ -0,0 +1,37 @@
+require "tsort"
+
+# Namespace for code related to /etc/fstab
+module Fstab
+  # Simple module implementing topological sorting for fstab entries
+  module TSort
+    extend Yast::Logger
+
+    # Sorts a list of fstab entries by the topological order of the mount points
+    def self.sort(list)
+      # Define the two iterators required by TSort
+      each_node = ->(&b) { list.reverse.each(&b) }
```

Comment:
Your solution didn't work because you did a copy&paste error and I pasted it blindly. What you really meant was:
```ruby
each_node = list.reverse.method(:each)
```
That is, without the lambda. And you are right, I like it more than the one using ```reversed```.